sklearn/ensemble/_hist_gradient_boosting/_predictor.pxd

```diff
@@ -0,0 +1,13 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: language_level=3
+# distutils: language=c++
+
+from .common cimport node_struct
+from libcpp.vector cimport vector
+
+cdef class TreePredictor:
+    cdef vector[node_struct] _nodes
+    cdef node_struct* get(self, int node_idx) nogil
+    cdef int get_size(self) nogil
```

Comment:
I think `get_n_nodes` might be a clearer name. In C, intuitively I read `get_size` as returning the size in bytes